examples/gno.land/r/demo/boards2/post.gno

```diff
@@ -97,6 +99,30 @@ func (post *Post) GetUpdatedAt() time.Time {
 	return post.updatedAt
 }
 
+func (post *Post) AddFlag(flag Flag) bool {
+	// TODO: sort flags for fast search in case of big thresholds
+	for _, v := range post.flags {
+		if v.User == flag.User {
+			return false
+		}
+	}
+
+	post.flags = append(post.flags, flag)
+	return true
+}
+
+func (post *Post) FlagsCount() int {
+	return len(post.flags)
+}
+
+func (post *Post) SetVisible(isVisible bool) {
+	post.isHidden = !isVisible
+}
+
+func (post *Post) Hidden() bool {
+	return post.isHidden
+}
```

Comment:
Maybe, for consistency, method name could be `IsHidden()` or the field `Post.hidden`:

```suggestion
func (post *Post) IsHidden() bool {
	return post.isHidden
}
```